src/RefractSourceMap.h

```diff
@@ -0,0 +1,73 @@
+//
+//  RefractSourceMap.h
+//  drafter
+//
+//  Created by Jiri Kratochvil on 31/07/15.
+//  Copyright (c) 2015 Apiary Inc. All rights reserved.
+//
+
+#ifndef DRAFTER_REFRACTSOURCEMAP_H
+#define DRAFTER_REFRACTSOURCEMAP_H
+
+#include "Serialize.h"
+
+namespace drafter {
+
+
+    refract::IElement* BytesRangeToRefract(const mdp::BytesRange& bytesRange);
+
+    template<typename T>
+    refract::IElement* SourceMapToRefract(const T& sourceMap) 
```

Comment:
Ah sorry, I did not read carefully I understand, you want to be visible just declarations of functions.

About "there is just 1 type", no every one `SourceMap<>` is indvidual type, but you are right the one comon ancestor.

Will check what can I use from this, while I start work on SourceMaps It was not accessible in drafter